### ISSUE TITLE
Clippy perceives the needless_return rule as invalid

### ISSUE THREAD

**Issue by muturgan** [rust-lang/rust-clippy#10637]
### Summary

I have added a `.clippy.toml` file to the root folder of my project.
With a single rule: `needless_return = true`.
https://rust-lang.github.io/rust-clippy/master/#needless_return

Then I try to execute a command:
`cargo clippy --all --all-features --tests -- -D warnings`
I expected to see a successful job finishing.

But I have an error:
```
error reading Clippy's configuration file `/home/<<username>>/<<path-to-project>>/.clippy.toml`: unknown field `needless_return`, expected one of
```
below is a large list of valid rules

### Reproducer

This is the .clippy.toml file

```
needless_return = true
```

I tried this code:

```rust
fn main() {
	five();
}

fn five() -> usize {
	return 5;
}
```

### Version

```text
rustc 1.68.2 (9eb3afe9e 2023-03-27)
binary: rustc
commit-hash: 9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0
commit-date: 2023-03-27
host: x86_64-unknown-linux-gnu
release: 1.68.2
LLVM version: 15.0.6
```

**Comment by Alexendoo** [comment]
The clippy config file is for some lints that have options they understand, it can't enable/disable lints - https://github.com/rust-lang/rust-clippy#configure-the-behavior-of-some-lints

`needless_return` warns by default, so you don't need to do anything to enable it. For those that aren't you'd typically do `#![warn(clippy::some_lint)]` or `#![warn(clippy::pedantic)]` for a bunch of them
